### ISSUE TITLE
False positive indent error for comment line before line started from semicolon

### ISSUE THREAD

**Issue by Alexsey** [eslint/eslint#12232]
**Tell us about your environment**

* **ESLint Version:** 6.3.0
* **Node Version:** 12.9.1
* **npm Version:** 6.11.3

**What parser (default, Babel-ESLint, etc.) are you using?** none

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  plugins: [
    'node',
    'lodash',
    'array-func',
    'import',
    'jest',
  ],
  extends: [
    'standard',
    'plugin:node/recommended',
  ],
  env: {
    node: true,
    es6: true,
    'jest/globals': true,
  },
  parserOptions: {
    'ecmaVersion': 9,
    'sourceType': 'script',
  },
  rules: {
    'comma-dangle': ['error', 'always-multiline'],
    'max-len': ['error', {'code': 120}],
    'no-tabs': 'error',

    'node/no-unpublished-require': 'off',
    'node/exports-style': ['error', 'module.exports'],

    'lodash/callback-binding': 'error',
    'lodash/collection-method-value': 'off',
    'lodash/collection-return': 'off',
    'lodash/no-double-unwrap': 'error',
    'lodash/no-extra-args': 'error',
    'lodash/no-unbound-this': 'error',
    'lodash/unwrap': 'error',
    'lodash/chain-style': ['error', 'as-needed'],
    'lodash/chaining': 'off',
    'lodash/consistent-compose': 'off',
    'lodash/identity-shorthand': ['error', 'always'],
    'lodash/import-scope': 'off',
    'lodash/matches-prop-shorthand': ['error', 'always'],
    'lodash/matches-shorthand': ['error', 'always'],
    'lodash/no-commit': 'off',
    'lodash/path-style': ['error', 'as-needed'],
    'lodash/prefer-compact': 'off',
    'lodash/prefer-filter': 'error',
    'lodash/prefer-find': 'error',
    'lodash/prefer-flat-map': 'error',
    'lodash/prefer-immutable-method': 'off',
    'lodash/prefer-invoke-map': 'error',
    'lodash/prefer-map': 'error',
    'lodash/prefer-reject': 'error',
    'lodash/prefer-thru': 'error',
    'lodash/prefer-wrapper-method': 'error',
    'lodash/preferred-alias': 'error',
    'lodash/prop-shorthand': ['error', 'always'],
    'lodash/prefer-constant': 'off',
    'lodash/prefer-get': 'error',
    'lodash/prefer-includes': 'error',
    'lodash/prefer-is-nil': 'error',
    'lodash/prefer-lodash-chain': 'off',
    'lodash/prefer-lodash-method': 'off',
    'lodash/prefer-matches': 'error',
    'lodash/prefer-noop': 'off',
    'lodash/prefer-over-quantifier': 'error',
    'lodash/prefer-some': 'error',
    'lodash/prefer-startswith': 'error',
    'lodash/prefer-times': 'error',

    'array-func/from-map': 'error',
    'array-func/no-unnecessary-this-arg': 'error',
    'array-func/prefer-array-from': 'off',
    'array-func/avoid-reverse': 'error',
    'array-func/prefer-flat-map': 'error',
    'array-func/prefer-flat': 'error',

    'import/order': ['error', {"newlines-between": "always"}],
  },
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const {log} = console

// comment <- error on this line

;(async () => {
  log('stuff')
})()
```

Autorun by IntelliJ IDE

**What did you expect to happen?**
No error

**What actually happened? Please include the actual, raw output from ESLint.**
ESLint: expected indentation of 2 spaces but found 0.(indent)

**Are you willing to submit a pull request to fix this bug?**
Probably no

**Comment by kaicataldo** [comment]
Verified using [our demo](https://eslint.org/demo#eyJ0ZXh0IjoiY29uc3Qge2xvZ30gPSBjb25zb2xlXG5cbi8vIGNvbW1lbnQgPC0gZXJyb3Igb24gdGhpcyBsaW5lXG5cbjsoYXN5bmMgKCkgPT4ge1xuICAgIGxvZygnc3R1ZmYnKVxufSkoKSIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6MTEsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiY29uc3RydWN0b3Itc3VwZXIiOjIsImZvci1kaXJlY3Rpb24iOjIsImdldHRlci1yZXR1cm4iOjIsIm5vLWFzeW5jLXByb21pc2UtZXhlY3V0b3IiOjIsIm5vLWNhc2UtZGVjbGFyYXRpb25zIjoyLCJuby1jbGFzcy1hc3NpZ24iOjIsIm5vLWNvbXBhcmUtbmVnLXplcm8iOjIsIm5vLWNvbmQtYXNzaWduIjoyLCJuby1jb25zdC1hc3NpZ24iOjIsIm5vLWNvbnN0YW50LWNvbmRpdGlvbiI6Miwibm8tY29udHJvbC1yZWdleCI6Miwibm8tZGVidWdnZXIiOjIsIm5vLWRlbGV0ZS12YXIiOjIsIm5vLWR1cGUtYXJncyI6Miwibm8tZHVwZS1jbGFzcy1tZW1iZXJzIjoyLCJuby1kdXBlLWtleXMiOjIsIm5vLWR1cGxpY2F0ZS1jYXNlIjoyLCJuby1lbXB0eSI6Miwibm8tZW1wdHktY2hhcmFjdGVyLWNsYXNzIjoyLCJuby1lbXB0eS1wYXR0ZXJuIjoyLCJuby1leC1hc3NpZ24iOjIsIm5vLWV4dHJhLWJvb2xlYW4tY2FzdCI6Miwibm8tZXh0cmEtc2VtaSI6Miwibm8tZmFsbHRocm91Z2giOjIsIm5vLWZ1bmMtYXNzaWduIjoyLCJuby1nbG9iYWwtYXNzaWduIjoyLCJuby1pbm5lci1kZWNsYXJhdGlvbnMiOjIsIm5vLWludmFsaWQtcmVnZXhwIjoyLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6Miwibm8tbWlzbGVhZGluZy1jaGFyYWN0ZXItY2xhc3MiOjIsIm5vLW1peGVkLXNwYWNlcy1hbmQtdGFicyI6Miwibm8tbmV3LXN5bWJvbCI6Miwibm8tb2JqLWNhbGxzIjoyLCJuby1vY3RhbCI6Miwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjoyLCJuby1yZWRlY2xhcmUiOjIsIm5vLXJlZ2V4LXNwYWNlcyI6Miwibm8tc2VsZi1hc3NpZ24iOjIsIm5vLXNoYWRvdy1yZXN0cmljdGVkLW5hbWVzIjoyLCJuby1zcGFyc2UtYXJyYXlzIjoyLCJuby10aGlzLWJlZm9yZS1zdXBlciI6Miwibm8tdW5leHBlY3RlZC1tdWx0aWxpbmUiOjIsIm5vLXVucmVhY2hhYmxlIjoyLCJuby11bnNhZmUtZmluYWxseSI6Miwibm8tdW5zYWZlLW5lZ2F0aW9uIjoyLCJuby11bnVzZWQtbGFiZWxzIjoyLCJuby11bnVzZWQtdmFycyI6Miwibm8tdXNlbGVzcy1jYXRjaCI6Miwibm8tdXNlbGVzcy1lc2NhcGUiOjIsIm5vLXdpdGgiOjIsInJlcXVpcmUtYXRvbWljLXVwZGF0ZXMiOjIsInJlcXVpcmUteWllbGQiOjIsInVzZS1pc25hbiI6MiwidmFsaWQtdHlwZW9mIjoyLCJpbmRlbnQiOjJ9LCJlbnYiOnt9fX0=).
